lib/puppetlabs_spec_helper/rake_tasks.rb

```diff
@@ -171,6 +171,7 @@
 # Must clear as it will not override the existing puppet-lint rake task since we require to import for
 # the PuppetLint::RakeTask
 Rake::Task[:lint].clear
+PuppetLint.configuration.fail_on_warnings = true
```

Comment:
I'm :100: to enable this. But I've no idea how this relates to
* https://github.com/puppetlabs/pdk-templates/blob/main/moduleroot/Rakefile.erb#L70
* https://github.com/puppetlabs/pdk-templates/blob/main/moduleroot/.puppet-lint.rc.erb#L18